File: 15-file_io.py
# pathlib 將 path 當物件處理，讀寫檔案會幾直觀。
from pathlib import Path


path = Path("tmp-python-file.txt")

path.write_text("Ada\nGrace\nLinus\n")

lines = path.read_text().splitlines()
uppercased = [line.upper() for line in lines]

print(lines)
print(uppercased)

print("---")

# with 會自動 close file，適合需要手動 open 嘅情況。
with path.open("a") as file:
    file.write("Matz\n")

for line in path.read_text().splitlines():
    print(f"- {line}")

path.unlink()

File: 15-file_io.rb
# Ruby File API 可以直接寫、讀、append，再記得清理臨時檔。
path = 'tmp-ruby-file.txt'

File.write(path, "Ada\nGrace\nLinus\n")

lines = File.readlines(path, chomp: true)
uppercased = lines.map(&:upcase)

p lines
p uppercased

puts '---'

# block 版 File.open 會自動 close file。
File.open(path, 'a') do |file|
  file.write("Matz\n")
end

File.readlines(path, chomp: true).each do |line|
  puts "- #{line}"
end

File.delete(path)
